### ISSUE TITLE
[Regression] snapshot is not changing language of Simulator

### ISSUE THREAD

**Issue by tobihagemann** [fastlane/fastlane#8895]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I can't really tell if this issue is new with Xcode 8.3 or some fastlane update. The last time it worked was with Xcode 8.2.1 and fastlane 2.22.0, as far as I can remember. The language of the Simulator stays `en-US` in my case and doesn't change at all for snapshots that are supposed to be done in other languages.

I tried resetting all simulators with `fastlane snapshot reset_simulators`, but that didn't help.

##### Complete output when running fastlane, including the stack trace and command used
```
$ fastlane snapshot
[14:19:54]: Successfully loaded '/Users/<redacted>/fastlane/Snapfile' 📄

+----------------------------+---------------------------------+
|          Detected Values from './fastlane/Snapfile'          |
+----------------------------+---------------------------------+
| scheme                     | <redacted>UITests               |
| devices                    | ["iPhone 7 Plus", "iPad Air 2"] |
| languages                  | ["en-US", "de-DE", "nl", "ru"]  |
| output_directory           | ./fastlane/screenshots          |
| clear_previous_screenshots | false                           |
+----------------------------+---------------------------------+

[14:19:54]: $ xcodebuild -list -workspace ./<redacted>.xcworkspace
[14:19:55]: $ xcodebuild clean -showBuildSettings -workspace ./<redacted>.xcworkspace -scheme <redacted>UITests

+----------------------------+----------------------------------------------------------------+
|                                 Summary for snapshot 2.27.0                                 |
+----------------------------+----------------------------------------------------------------+
| scheme                     | <redacted>UITests                                              |
| devices                    | ["iPhone 7 Plus", "iPad Air 2"]                                |
| languages                  | ["en-US", "de-DE", "nl", "ru"]                                 |
| output_directory           | /Users/<redacted>/fastlane/screenshots                         |
| clear_previous_screenshots | false                                                          |
| workspace                  | ./<redacted>.xcworkspace                                       |
| launch_arguments           | [""]                                                           |
| output_simulator_logs      | false                                                          |
| skip_open_summary          | false                                                          |
| skip_helper_version_check  | false                                                          |
| reinstall_app              | false                                                          |
| erase_simulator            | false                                                          |
| localize_simulator         | false                                                          |
| app_identifier             | <redacted>                                                     |
| buildlog_path              | ~/Library/Logs/snapshot                                        |
| clean                      | false                                                          |
| number_of_retries          | 1                                                              |
| stop_after_first_error     | false                                                          |
| xcode_path                 | /Applications/Xcode.app                                        |
+----------------------------+----------------------------------------------------------------+

[14:19:56]: Building and running project - this might take some time...
[14:19:56]: snapshot run 1 of 8
[14:19:56]: Patching '/Users/<redacted>/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[14:19:57]: -----------------------------
[14:19:57]: --- iPhone 7 Plus - en-US ---
[14:19:57]: -----------------------------
[14:19:57]: $ set -o pipefail && xcodebuild -workspace ./<redacted>.xcworkspace -scheme <redacted>UITests -derivedDataPath '/var/folders/rq/yh4xk5vj7618tfvk_3wb40z00000gn/T/snapshot_derived20170416-86737-5kkiuh' -destination 'platform=iOS Simulator,id=A6678760-BE76-4C26-AE71-04123B8E1DAE,OS=10.3' FASTLANE_SNAPSHOT=YES build test | tee /Users/<redacted>/Library/Logs/snapshot/<redacted>UITests-<redacted>UITests.log | xcpretty 
...
[14:21:15]: Running Tests: ▸ Touching <redacted>.app
[14:21:15]: ▸ Signing /var/folders/rq/yh4xk5vj7618tfvk_3wb40z00000gn/T/snapshot_derived20170416-86737-5kkiuh/Build/Products/Debug-iphonesimulator/<redacted>.app
[14:21:16]: ▸ Building <redacted>/<redacted>UITests [Debug]
[14:21:16]: ▸ Check Dependencies
[14:21:53]: ▸ Test Succeeded
[14:21:53]: ▸ All tests
[14:21:53]: ▸ Test Suite <redacted>UITests.xctest started
[14:21:53]: ▸ <redacted>UITests
[14:21:53]: ▸ ✓ testSnapshot (24.318 seconds)
[14:21:53]: ▸ 	 Executed 1 test, with 0 failures (0 unexpected) in 24.318 (24.320) seconds
[14:21:53]: ▸ 
[14:21:54]: Collecting screenshots...
[14:21:54]: Found 5 screenshots...
[14:21:54]: Copying '/Users/<redacted>/fastlane/screenshots/en-US/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:21:54]: Copying '/Users/<redacted>/fastlane/screenshots/en-US/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:21:54]: Copying '/Users/<redacted>/fastlane/screenshots/en-US/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:21:54]: Copying '/Users/<redacted>/fastlane/screenshots/en-US/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:21:54]: Copying '/Users/<redacted>/fastlane/screenshots/en-US/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:21:54]: snapshot run 2 of 8
[14:21:54]: Patching '/Users/<redacted>/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[14:21:55]: -----------------------------
[14:21:55]: --- iPhone 7 Plus - de-DE ---
[14:21:55]: -----------------------------
[14:21:55]: $ set -o pipefail && xcodebuild -workspace ./<redacted>.xcworkspace -scheme <redacted>UITests -derivedDataPath '/var/folders/rq/yh4xk5vj7618tfvk_3wb40z00000gn/T/snapshot_derived20170416-86737-5kkiuh' -destination 'platform=iOS Simulator,id=A6678760-BE76-4C26-AE71-04123B8E1DAE,OS=10.3' FASTLANE_SNAPSHOT=YES build test | tee /Users/<redacted>/Library/Logs/snapshot/<redacted>UITests-<redacted>UITests.log | xcpretty 
...
[14:22:04]: Running Tests: ▸ Touching <redacted>.app
[14:22:04]: ▸ Signing /var/folders/rq/yh4xk5vj7618tfvk_3wb40z00000gn/T/snapshot_derived20170416-86737-5kkiuh/Build/Products/Debug-iphonesimulator/<redacted>.app
[14:22:04]: ▸ Building <redacted>/<redacted>UITests [Debug]
[14:22:04]: ▸ Check Dependencies
[14:22:48]: ▸ Test Succeeded
[14:22:48]: ▸ All tests
[14:22:48]: ▸ Test Suite <redacted>UITests.xctest started
[14:22:48]: ▸ <redacted>UITests
[14:22:48]: ▸ ✓ testSnapshot (23.762 seconds)
[14:22:48]: ▸ 	 Executed 1 test, with 0 failures (0 unexpected) in 23.762 (23.763) seconds
[14:22:48]: ▸ 
[14:22:48]: Collecting screenshots...
[14:22:48]: Found 5 screenshots...
[14:22:48]: Copying '/Users/<redacted>/fastlane/screenshots/de-DE/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:22:48]: Copying '/Users/<redacted>/fastlane/screenshots/de-DE/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:22:48]: Copying '/Users/<redacted>/fastlane/screenshots/de-DE/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:22:48]: Copying '/Users/<redacted>/fastlane/screenshots/de-DE/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:22:48]: Copying '/Users/<redacted>/fastlane/screenshots/de-DE/iPhone7Plus-<redacted>-d41d8cd98f00b204e9800998ecf8427e.png'...
[14:22:48]: snapshot run 3 of 8
[14:22:49]: Patching '/Users/<redacted>/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
...

+---------------+-------+-------+-----+-----+
|             snapshot results              |
+---------------+-------+-------+-----+-----+
| Device        | en-US | de-DE | nl  | ru  |
+---------------+-------+-------+-----+-----+
| iPhone 7 Plus |  💚   |  💚   |  💚 |  💚 |
| iPad Air 2    |  💚   |  💚   |  💚 |  💚 |
+---------------+-------+-------+-----+-----+

[14:28:40]: Generating HTML Report
[14:28:40]: Successfully created HTML file with an overview of all the screenshots: '/Users/<redacted>/fastlane/screenshots/screenshots.html'
```

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.4                                     |
| Ruby                        | 2.4.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.rvm/gems/ruby-head/bin/fastlane          |
| Host                        | Mac OS X 10.12.4 (16E195)                   |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-head/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2h  3 May 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.1                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Snapfile`</summary>

```ruby
scheme "<redacted>UITests"

devices([
  "iPhone 7 Plus",
  "iPad Air 2"
])

languages([
  "en-US",
  "de-DE",
  "nl",
  "ru"
])

output_directory './fastlane/screenshots'

clear_previous_screenshots false
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.27.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.13.6       |
| io-console                | 0.4.6        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.20       |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| faraday_middleware        | 0.10.1       |
| rubyzip                   | 1.2.0        |
| plist                     | 3.2.0        |
| fastimage                 | 2.0.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
</details>

*generated on:* **2017-04-16**
</details>

**Comment by lukas2** [comment]
Confirm, also downgrading to 2.22.0 fixed the problem for me.

**Comment by KrauseFx** [comment]
There was no changes in that regard in _snapshot_ in the most recent release. Do you know with what version that issue was introduced?

**Comment by tobihagemann** [comment]
Yes, I've pretty quickly nailed down the version, because I suspected the new `SnapshotHelper.swift` [1.3] is not working.

fastlane 2.23.0 works for me. Beginning with fastlane 2.24.0, the language of Simulator doesn't change anymore.

Maybe it doesn't have to do with the `SnapshotHelper` itself, it was just a wild guess. If there's something else that I should try out, I'd be happy to help!

**Comment by tobihagemann** [comment]
I think it's very likely that there is a bug in `SnapshotHelper.swift`. I just updated to fastlane 2.27.0, but kept using the old helper. Basically, I faked the version number by putting in `// SnapshotHelperVersion [1.3]`. Maybe that's a hint?

**Comment by Overbored** [comment]
To make it work I replaced row 144-148 in SnapshotHelper.swift:
            guard let homeDirUrl = URL(string: simulatorHostHome) else {
                print("Can't prepare environment. Simulator home location is inaccessible. Does \(simulatorHostHome) exist?")
                return nil
            }

with a single row:
            let homeDirUrl = URL(fileURLWithPath: simulatorHostHome)

Don't know why the URL(string... initializer doesn't work but it only returns nil if the URL can't be formed. It does not check if location exist.

**Comment by backmeupplz** [comment]
Same here, version 2.27.0, simulator runs only in `en-US`. Fix provided by @Overbored resolved the issue

**Comment by wryphonedev** [comment]
Duplicated issue: #8768 #8766 #8733 #8755
This issue was also addressed in PR #8791

**Comment by KrauseFx** [comment]
This issue is now resolved with the most recent release, right?
